libvast/test/parseable.cpp

```diff
@@ -595,6 +597,50 @@ TEST(real) {
   //  CHECK(f == str.begin() + 4);
 }
 
+TEST(real - scientific) {
+  auto p = make_parser<real>{};
+  {
+    MESSAGE("null exponent");
+    auto str = ".456789e0"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 0.456789);
+    CHECK_EQUAL(f, l);
+  }
+  {
+    MESSAGE("positive exponent");
+    auto str = ".456789e43"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 4.56789e42);
+    CHECK_EQUAL(f, l);
+  }
+  {
+    MESSAGE("explicit positive exponent");
+    auto str = ".456789e+43"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 4.56789e42);
```

Comment:
Why does the string end in 43 and the C++ literal in 42? (Also for other tests)

Reply:
The dot moved one position to the right.